cli/error.go

```diff
@@ -6,16 +6,22 @@ import (
 
 // StatusError reports an unsuccessful exit by a command.
 type StatusError struct {
-	Status     string
+	Status     error
 	StatusCode int
 }
 
 // Error formats the error for printing. If a custom Status is provided,
 // it is returned as-is, otherwise it generates a generic error-message
 // based on the StatusCode.
 func (e StatusError) Error() string {
-	if e.Status == "" {
+	if e.Status == nil {
 		return "exit status " + strconv.Itoa(e.StatusCode)
 	}
+	return e.Status.Error()
+}
+
+// Unwrap allows wrapped errors stored in StatusError to be checked
+// using `errors.Is`.
```

Comment:
`Unwrap` doesn't allow wrapped errors to be checked using `errors.Is` – it allows `StatusError` to be checked with `errors.Is` (and exposes the underlying error).
```suggestion
// Unwrap returns the wrapped error.
// 
// This allows StatusError to be checked with errors.Is.
```

Reply:
True, what I tried to say was: `errors.Is` checks for errors down the error chain. Since we wrap errors using `fmt.Errorf("%w", err)` and store that inside `StatusError` we need to check the underlying error instead of just `err == StatusError`.

Reply:
Right, it's just a matter of wording. IMO the GoDoc for `Unwrap` should explain what (from the perspective of `StatusError`) what `Unwrap` does, which specifically is "return the wrapped error – which allows checking StatusErr with `errors.Is`".

`allows wrapped errors stored in StatusError to be checked using errors.Is.` is less clear to me since it could also be interpreted as "allows you to do `errors.Is` on the underlying error" which isn't the case.